### PR TITLE
refactor[v2]: fixing revive issues to make project revive compatible

### LINKED ISSUE #1054: Make api/v1alpha1 revive compatible
KEDA uses [golint](https://github.com/golang/lint) to ensure code quality. However, some files still need adjusting:

- [ ] api/v1alpha1/condition_types.go
- [ ] api/v1alpha1/groupversion_info.go
- [ ] api/v1alpha1/gvkr_types.go
- [ ] api/v1alpha1/scaledjob_types.go
- [ ] api/v1alpha1/scaledobject_types.go
- [ ] api/v1alpha1/triggerauthentication_types.go

### How 

1. Create a new branch from `v2`branch
2. Remove the files from list in [Makefile](https://github.com/kedacore/keda/blob/5fe6c52ad38a44726b74e1339e7197e193d1c26c/Makefile#L212-L242)
3. Run `make revive`
4. Commit your changes and open a PR to `v2`

### Reference
https://github.com/kedacore/keda/pull/1025#discussion_r475134492

### PR DESCRIPTION
**Pull request by aman-bansal**
Recently go lint has been changed to revive. There were pending linting issues in few files. This PR is to address those issues.
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #1054

### CODE REVIEW

File: pkg/scaling/executor/scale_executor.go [hunk 1]
```diff
@@ -31,6 +32,7 @@ type scaleExecutor struct {
 	logger           logr.Logger
 }
 
+//NewScaleExecutor creates a ScaleExecutor object
```
Comment (ahmelsayed) [inline pkg/scaling/executor/scale_executor.go:35]:
```suggestion
// NewScaleExecutor creates a ScaleExecutor object
```

File: api/v1alpha1/triggerauthentication_types.go [hunk 2]
```diff
@@ -46,6 +46,8 @@ type TriggerAuthenticationList struct {
 // PodIdentityProvider contains the list of providers
 type PodIdentityProvider string
 
+// PodIdentityProviderNone specifies the default state when there is no Identity Provider
+//PodIdentityProvider<IDENTITY_PROVIDER> specifies other available Identity providers
```
Comment (ahmelsayed) [inline api/v1alpha1/triggerauthentication_types.go:50]: minor nit
```suggestion
// PodIdentityProvider<IDENTITY_PROVIDER> specifies other available Identity providers
```

File: pkg/scaling/resolver/scale_resolvers.go [hunk 3]
```diff
@@ -12,6 +12,8 @@ import (
 	kedav1alpha1 "github.com/kedacore/keda/api/v1alpha1"
 )
 
+// ResolveContainerEnv resolves all environment variables in a container.
+//It returns either map of env variable key and value or error if there is any.
```
Comment (ahmelsayed) [inline pkg/scaling/resolver/scale_resolvers.go:16]:
```suggestion
// It returns either map of env variable key and value or error if there is any.
```

**Comment by ahmelsayed** [comment]
Thanks @aman-bansal for taking care of this.

**Comment by aman-bansal** [comment]
@ahmelsayed all changes done.

**Review by ahmelsayed — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Comment by ahmelsayed** [comment]
Thanks @turbaszek

**Pull request merged**